src/main/java/com/squareup/kotlinpoet/Parameter.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.squareup.kotlinpoet
+
+import java.lang.reflect.Type
+import kotlin.reflect.KClass
+
+data class Parameter(val name: String? = null, val type: TypeName) {
+
+  constructor(name: String, type: KClass<*>) : this(name, type.asTypeName())
+
+  constructor(name: String, type: Type) : this(name, type.asTypeName())
+
+  internal fun toCodeBlock() = if (name != null) {
```

Comment:
off topic: would it be better to do this kind of expression without braces?
```
internal fun toCodeBlock() = if (name != null)
    CodeBlock.of("%L: %T", name, type) else
    CodeBlock.of("%T", type)
```

Reply:
Yeah, I think that'd be fine with the guidelines